index.js

```diff
@@ -128,7 +128,7 @@ var appServer = function(config) {
         postRequest: self.config.postRequest
       });
 
-      var endpoint = path.posix.join(normalizedRoot, pkg.name);
+      var endpoint = path.posix.join(normalizedRoot, app.name);
       self.log("   loaded app [" + pkg.name + "] at endpoint: " + endpoint);
```

Comment:
So I added the test, with a new entry in the "invalid_examples" folder, and added the assertion in the existing invalid examples test. I wasn't sure where it fit the best, so I'm happy to restructure if you think there's a better spot for it.